scIB/metrics.py

```diff
@@ -41,27 +41,34 @@ def __init__(self, message):
  
 
 ### Silhouette score
-def silhouette(adata, group_key, metric='euclidean', embed='X_pca', scale=True):
+def silhouette(adata, group_key, embed, metric='euclidean', scale=True):
     """
-    wrapper for sklearn silhouette function values range from [-1, 1] with 1 being an ideal fit, 0 indicating overlapping clusters and -1 indicating misclassified cells
+    wrapper for sklearn silhouette function values range from [-1, 1] with 1 being an ideal fit, 0 indicating
+    overlapping clusters and -1 indicating misclassified cells
+    :param group_key: key in adata.obs of cell labels
+    :param embed: embedding key in adata.obsm, usually 'X_pca'
     """
     if embed not in adata.obsm.keys():
         print(adata.obsm.keys())
         raise KeyError(f'{embed} not in obsm')
-    asw = sklearn.metrics.silhouette_score(adata.obsm[embed], adata.obs[group_key], metric=metric)
+    asw = sklearn.metrics.silhouette_score(
+        X=adata.obsm[embed],
+        labels=adata.obs[group_key],
+        metric=metric
+    )
     if scale:
         asw = (asw + 1)/2
     return asw
 
-def silhouette_batch(adata, batch_key, group_key, metric='euclidean', 
-                     embed='X_pca', verbose=True, scale=True):
+def silhouette_batch(adata, batch_key, group_key, embed, metric='euclidean',
+                     verbose=True, scale=True):
```

Comment:
To keep this clean, we need to adapt how this is called in the metrics function if you change the order of parameters.

Reply:
The way we currently have it in the `metrics` function is 
```
sil_global = silhouette(adata_int, group_key=label_key, embed=embed, metric=si_metric)
```
So no changes needed.

Reply:
This is for the cluster ASW, but not for the `silhouette_batch()` call if I'm not mistaken. There you call these arguments in a different order i believe.

Reply:
The arguments are defined and passed as `silhouette_batch(adata, batch_key, group_key, embed)`

Reply:
hmm, i seem to have misread. You're right.